code/src/java/pcgen/util/FileHelper.java

```diff
@@ -117,4 +117,22 @@ private static String stripOffRoot(File relative)
 
 		return relative.getAbsolutePath().substring(root.length(), relative.getAbsolutePath().length());
 	}
+
+	/**
+	 * Remove characters which are not allowed by certain OSes and filesystems. The string returned can be used
+	 * as a filename on all OSes which can run pcgen.
+	 *
+	 * The following characters are removed:
+	 * <ul>
+	 *     <li>&lt;, &gt; :, ", \\, |, ?, * because they are
+	 *     <a href="https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#naming-conventions">restricted on Windows</a></li>
+	 *     <li>/ and U+0000 (Unicode NULL) , restricted on Linux, Mac, and Windows</li>
+	 * </ul>
+	 *
+	 * @param invalidCharacterFilename Filename to sanitize
+	 * @return string with invalid characters removed
+	 */
+	public static String sanitizeFilename(String invalidCharacterFilename) {
+		return invalidCharacterFilename.replaceAll("[\u0000\\\\/:*?\"<>|]", "");
```

Comment:
Null check for the parameter? Recently I started using Lombok NonNull annotation for parameters.

Reply:
Good point. Will adjust.

Reply:
I'm checking other parts of the code to see if there is a pattern to parameter checking. Does anyone know the preferred pattern?

We could add lombok, but I'm not sure I want to add a dependency in my first PR. 

Another option would be to do an explicit check and throw a null pointer exception with clear messaging (similar to what lombok would do with @NotNull.

Option 3 would be to check for null, send something to the system logs, but return an empty string as a safe value for downstream code.